kf_model_fhir/mappers/resources/kfdrc_vital_status.py

```diff
@@ -0,0 +1,78 @@
+"""
+This module converts Kids First Outcomes to FHIR kfdrc-vital-status
+(derived from FHIR Observation).
+"""
+
+from kf_lib_data_ingest.common import constants
+from kf_lib_data_ingest.common.concept_schema import CONCEPT
+from common.utils import make_identifier, make_select, get
+
+RESOURCE_TYPE = 'Observation'
+
+interpretation = {
+    constants.OUTCOME.VITAL_STATUS.ALIVE: True,
+    constants.OUTCOME.VITAL_STATUS.DEAD: False,
+    constants.COMMON.UNKNOWN: {
+        'system': 'http://terminology.hl7.org/CodeSystem/v3-NullFlavor',
+        'code': 'NI',
+        'display': 'No information'
+    }
+}
+
+def yield_kfdrc_vital_statuses(eng, table, study_id, kfdrc_patients):
+    for row in make_select(
+            eng, table,
+            CONCEPT.PARTICIPANT.ID,
+            CONCEPT.OUTCOME.EVENT_AGE_DAYS,
+            CONCEPT.OUTCOME.VITAL_STATUS
+        ):
+        participant_id = get(row, CONCEPT.PARTICIPANT.ID)
+        event_age_days = get(row, CONCEPT.OUTCOME.EVENT_AGE_DAYS)
+        vital_status = get(row, CONCEPT.OUTCOME.VITAL_STATUS)
+        status = interpretation.get(vital_status)
+
+        if (not participant_id or not status):
+            continue
+
+        retval = {
+            'resourceType': RESOURCE_TYPE,
+            'id': make_identifier(RESOURCE_TYPE, participant_id, vital_status, event_age_days),
+            'meta': {
+                'profile': [
+                    'http://fhir.kids-first.io/StructureDefinition/kfdrc-vital-status'
+                ]
+            },
+            'status': 'preliminary',
+            'code': {
+                'coding': [
+                    {
+                        'code': '438949009',
+                        'system': 'http://snomed.info/sct',
+                        'display': 'Alive'
+                    }
+                ],
+                'text': 'Alive'
+            },
```

Comment:
Shouldn't it be also dynamic?

Reply:
> Shouldn't it be also dynamic?

What do you mean? Are you asking if the `code` should change like the `value[x]` does?

Reply:
Talked about it with @gsantia, and observation code is for indicating a type of Observation (https://www.hl7.org/fhir/observation.html). `'Alive'` seems more like a value of a specific observation type. So, I recommended using SNOMED's [`'Clinical status'`]( https://browser.ihtsdotools.org/?perspective=full&conceptId1=263493007&edition=MAIN/2020-03-09&release=&languages=en) as a type.

Then, `interpretation` could be a dictionary of CodeableConcepts like this:

```python
clinical_status = {
    constants.OUTCOME.VITAL_STATUS.ALIVE: {
        'coding': [
            {
                'code': '438949009',
                'system': 'http://snomed.info/sct',
                'display': 'Alive'
            }
         ],
        'text': 'Alive'
    },
    constants.OUTCOME.VITAL_STATUS.DEAD: ...
}
```
Then, the conditional for valueCodeableConcept and valueBoolean could always be like

```python
if vital_status:
    retval['valueCodeableConcept'] = clinical_status[vital_status]
```

Reply:
> Talked about it with @gsantia, and observation code is for indicating a type of Observation (https://www.hl7.org/fhir/observation.html). `'Alive'` seems more like a value of a specific observation type. So, I recommended using SNOMED's [`'Clinical status'`](https://browser.ihtsdotools.org/?perspective=full&conceptId1=263493007&edition=MAIN/2020-03-09&release=&languages=en) as a type.
> 
> Then, `interpretation` could be a dictionary of CodeableConcepts like this:
> 
> ```python
> clinical_status = {
>     constants.OUTCOME.VITAL_STATUS.ALIVE: {
>         'coding': [
>             {
>                 'code': '438949009',
>                 'system': 'http://snomed.info/sct',
>                 'display': 'Alive'
>             }
>          ],
>         'text': 'Alive'
>     },
>     constants.OUTCOME.VITAL_STATUS.DEAD: ...
> }
> ```
> 
> Then, the conditional for valueCodeableConcept and valueBoolean could always be like
> 
> ```python
> if vital_status:
>     retval['valueCodeableConcept'] = clinical_status[vital_status]
> ```

Updated!

Reply:
Hmm I thought we were trying to represent something like this `vital-status.alive=[True|False]`. I think that's why the `Observation.code` was fixed to the `Alive` concept and we kept the `valueBoolean` to capture the boolean value of it. 

I guess you can do it either way? I wonder what @ShahimEssaid thinks?

Reply:
@znatty22, yeah I know we discussed that. But, [the above SNOMED code](https://browser.ihtsdotools.org/?perspective=full&conceptId1=438949009&edition=MAIN/2020-03-09&release=&languages=en) is for indicating a general clinical state of being alive. For external users, it may be confusing since Observation.code always says `"Alive"` even if a resource has `false` at `'valueBoolean'`.

Reply:
Using the snomed clinical status for the "code" would give a richer set of values for the "value[x]" field. It's a matter of choice, and I'm not sure if any FHIR community is already adopting a specific way for doing this.  I think "interpretation" isn't relevant in this case. It's meant for interpreting the meaning of a value[x] when that's needed.

Reply:
Ok sounds good 👍, let's use the codes in SNOMED to indicate the value of the vital-status.